hudi-client/src/main/java/org/apache/hudi/table/HoodieCommitArchiveLog.java

```diff
@@ -219,14 +220,23 @@ private boolean deleteArchivedInstants(List<HoodieInstant> archivedInstants) thr
    * @throws IOException in case of error
    */
   private boolean deleteAllInstantsOlderorEqualsInAuxMetaFolder(HoodieInstant thresholdInstant) throws IOException {
-    List<HoodieInstant> instants = metaClient.scanHoodieInstantsFromFileSystem(
-        new Path(metaClient.getMetaAuxiliaryPath()), HoodieActiveTimeline.VALID_EXTENSIONS_IN_ACTIVE_TIMELINE, false);
+    List<HoodieInstant> instants = null;
+    boolean success = true;
+    try {
+      instants =
+          metaClient.scanHoodieInstantsFromFileSystem(
+              new Path(metaClient.getMetaAuxiliaryPath()),
+              HoodieActiveTimeline.VALID_EXTENSIONS_IN_ACTIVE_TIMELINE,
+              false);
+    } catch (FileNotFoundException e) {
```

Comment:
Can you please provide specific pointers on how the folder gets removed on GCS? Is it GCS specific?

Reply:
not sure about it. Did see that folder disappears/reappears sometimes. Can be related to the way GCS connector treats a situation when all the files from the folder are removed. 

On gcs folders are not real, they are just logical. So, file with name: gs://bucket/folder/file doesn't live in the folder "folder", it just has a long "/" separated name. I saw some logic in the GCS FS java implementation that creates fake file "/" when everything is removed from the logical folder. So, my speculation is that if this call fails or not enabled, that removing all the files from the folder will end up if folder being removed as well.

Reply:
@afilipchik : Can you add the above comment in the Exception code block. Easy to understand the reason for this code. Also, Does this mean, we need to create the "aux" folder next time we create a file under aux ?

Reply:
Will add comment. On create folder -> no, as GCS will create "folder" automatically.